File: PLVisualizer.Api.Dto/Tables/Discipline.cs
namespace PlVisualizer.Api.Dto.Tables;

/// <summary>
/// Represents one item of distributed pedagogical load (one row in an Excel table).
/// </summary>
public record LoadDetail(string LoadType, int Hours, string Audience);

/// <summary>
/// Represents discipline
/// </summary>
public record Discipline(
    Guid Id,
    int TotalLoad,
    string Term,
    string Code,
    string Name,
    string GeneralWorkType,
    string Audience,
    IEnumerable<LoadDetail> LoadDetails
);

File: frontend/src/Models/Discipline.ts
import {Guid} from "guid-typescript";

export interface LoadDetails {
    loadType: string
    hours: number
    audience: string
}

export interface Discipline {
    id: Guid
    totalLoad : number
    term: string
    code: string
    name: string
    generalWorkType: string
    audience: string
    loadDetails: LoadDetails[]
}